kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/NettyFilterDispatchExecutorTest.java

```diff
@@ -111,7 +126,68 @@ void completeOnFilterDispatchThreadWhenInEventLoop() {
             CompletionStage<Boolean> inDispatchThread = dispatchExecutor.completeOnFilterDispatchThread(completionStage)
                     .thenApply(o -> dispatchExecutor.isInFilterDispatchThread());
             completionStage.complete("arbitrary");
-            assertThat(inDispatchThread).succeedsWithin(5, TimeUnit.SECONDS).isEqualTo(true);
+            assertThat(toStandardCompletableFuture(inDispatchThread)).succeedsWithin(5, TimeUnit.SECONDS).isEqualTo(true);
+        }
+        finally {
+            eventLoop.shutdownGracefully();
+        }
+    }
+
+    static <T> CompletableFuture<T> toStandardCompletableFuture(CompletionStage<T> stage) {
```

Comment:
toStandardCompletableFuture isn't a good name. chainStandardCompletableFuture would be more descriptive.